app/controllers/api/v1/admin/roles_controller.rb

```diff
@@ -17,6 +18,40 @@ def index
           roles = Role.select(:id, :name, :color)&.order(sort_config)&.search(params[:search])
           render_json data: roles
         end
+
+        # POST /api/v1/roles.json
+        # Expects: { role: {:name, :color} }
+        # Returns: { data: Array[serializable objects] , errors: Array[String] }
+        # Does: Creates a role.
+
+        def create
+          role = Role.new role_params
+
+          return render_json errors: role.errors.to_a, status: :bad_request unless role.save
```

Comment:
Although this works. I prefer the simpler (easier to read) way that we have in other controllers such as in rooms_controller

Reply:
Which is?

Reply:
Is it the  `render_error`?
If that's the case then this PR was made way before including it so I forgot to include it, will do Inshallah.

Reply:
It isn't a big deal and you don't need to change it. I am just trying to keep simple CRUD procedures following the same format is possible